starlingcaptureapi/fs_watcher.py

```diff
@@ -27,106 +29,163 @@ def caught_and_logged_exceptions(event):
+        return self
+
+
+class AddHandler(OrganizationHandler):
+    """Handles file changes for add action."""
+
+    def on_created(self, event):
+        with caught_and_logged_exceptions(event):
+            _actions.add(event.src_path, self.org_config, self.collection_id)
+
+
+class UpdateHandler(OrganizationHandler):
+    """Handles file changes for update action."""
 
-            Args:
-                organization_id: string with the unique organization id this handler is for
+    def on_created(self, event):
+        with caught_and_logged_exceptions(event):
+            _actions.update(event.src_path, self.org_config, self.collection_id)
 
-            Returns:
-                the handler itself
-            """
-            self.organization_id = organization_id
-            return self
 
-    class AddHandler(OrganizationHandler):
-        """Handles file changes for add action."""
+class StoreHandler(OrganizationHandler):
+    """Handles file changes for store action."""
 
-        def on_created(self, event):
-            with caught_and_logged_exceptions(event):
-                _actions.add(self.organization_id, event.src_path)
+    def on_created(self, event):
+        with caught_and_logged_exceptions(event):
+            _actions.store(event.src_path, self.org_config, self.collection_id)
 
-    class UpdateHandler(OrganizationHandler):
-        """Handles file changes for update action."""
 
-        def on_created(self, event):
-            with caught_and_logged_exceptions(event):
-                _actions.update(self.organization_id, event.src_path)
+class CustomHandler(OrganizationHandler):
+    """Handles file changes for custom action."""
 
-    class StoreHandler(OrganizationHandler):
-        """Handles file changes for store action."""
+    def on_created(self, event):
+        with caught_and_logged_exceptions(event):
+            _actions.custom(event.src_path, self.org_config, self.collection_id)
 
-        def on_created(self, event):
-            with caught_and_logged_exceptions(event):
-                _actions.store(self.organization_id, event.src_path)
 
-    class CustomHandler(OrganizationHandler):
-        """Handles file changes for custom action."""
+class ArchiveHandler(OrganizationHandler):
+    """Handles file changes for Archive action."""
 
-        def on_created(self, event):
-            with caught_and_logged_exceptions(event):
-                _actions.custom(self.organization_id, event.src_path)
+    def on_created(self, event):
+        with caught_and_logged_exceptions(event):
+            _actions.archive(event.src_path, self.org_config, self.collection_id)
 
-    class ArchiveHandler(OrganizationHandler):
-        """Handles file changes for Archive action."""
 
-        def on_created(self, event):
-            with caught_and_logged_exceptions(event):
-                _actions.archive(event.src_path)
+# Mapping from action name to handler class
+ACTION_HANDLER = {
```

Comment:
Add `c2pa-` to all these except `archive`.

Reply:
Done.